File: practica2_820574_839304/java/Read.java
/*
* Fichero: Read.java
* Autores: Alicia Lázaro Huerta y Manel Jordá Puig Rubio
* Fecha: 27/02/2023
*/
import java.util.Scanner;
import java.util.Stack;

public class Read extends Instruccion {
    Read(){
        nombre = "read";
    }
    public void ejecutar(Stack<Integer> pila, int[] pc){
        Scanner scanner = new Scanner(System.in);
        int a;
        System.out.print("? ");
        a = scanner.nextInt();
        pila.push(a);
        pc[0]++;
    }
}

File: practica2_820574_839304/c++/read.c
#include "read.h"

Read::read() {
    nombre = "read";
}

void Read::ejecutar(PilaInts &pila, int &pc) {
    int a;
    cin >> a;
    pila.insertar(a);
}

string Read::nombrar() {
    return nombre;
}